Kudu.Core/Settings/DiagnosticsSettingsManager.cs

```diff
@@ -87,6 +93,7 @@ private void SaveSettings(DiagnosticsSettings diagnosticsSettings)
                 FileSystemHelpers.EnsureDirectory(Path.GetDirectoryName(_path));
             }
 
+            diagnosticsSettings.EnsureValid();
             string fileContent = JsonConvert.SerializeObject(diagnosticsSettings);
```

Comment:
I also reviewed Azure codes and still couldn't find how this could have happenned from the normal diagnostics api usages.